horace_core/sqw/@sqw/cut.m

```diff
@@ -136,32 +93,30 @@
 %
 
 if numel(obj) > 1
-    error('HORACE:cut:invalid_argument', ...
-        ['You cannot take a cut from an array, or cell array, of sqw or ' ...
-        'dnd objects.\nConsider cutting the objects in a loop.']);
+    wout = cut(num2cell(obj),varargin{:});
+    return;
 end
 hc= hor_config;
 log_level = hc.log_level;
 
 dnd_type = obj.pix.num_pixels == 0;
 if dnd_type
-    % Input has no pixels, delegate to cut_dnd
-    % TODO: refactor so cut_dnd_main sits on DnDBase class
-    ndims_source = numel(obj.data.pax);
-    wout = cut_dnd_main(obj, ndims_source, varargin{:});
+    % Input has no pixels, delegate to cut_dnd as the result is dnd
+    wout = cut(obj.data,varargin{:});
```

Comment:
does not appear to be a delegation to cut_dnd but to some cut. Need explanation or change of function

Reply:
this is how inheritance works. Call appropriate class method delegates the call to this class